x-pack/plugins/ml/server/plugin.ts

```diff
@@ -0,0 +1,168 @@
+    this.version = ctx.env.packageInfo.branch;
+  }
+
+  public setup(coreSetup: CoreSetup, plugins: PluginsSetup) {
+    let sampleLinksInitialized = false;
+
+    plugins.features.registerFeature({
+      id: PLUGIN_ID,
+      name: i18n.translate('xpack.ml.featureRegistry.mlFeatureName', {
+        defaultMessage: 'Machine Learning',
+      }),
+      icon: 'machineLearningApp',
+      navLinkId: PLUGIN_ID,
+      app: [PLUGIN_ID, 'kibana'],
+      catalogue: [PLUGIN_ID],
+      privileges: {},
+      reserved: {
+        privilege: {
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          ui: [],
+        },
+        description: i18n.translate('xpack.ml.feature.reserved.description', {
+          defaultMessage:
+            'To grant users access, you should also assign either the machine_learning_user or machine_learning_admin role.',
+        }),
+      },
+    });
+
+    // Can access via router's handler function 'context' parameter - context.ml.mlClient
+    const mlClient = coreSetup.elasticsearch.createClient(PLUGIN_ID, {
+      plugins: [elasticsearchJsPlugin],
+    });
+
+    coreSetup.http.registerRouteHandlerContext(PLUGIN_ID, (context, request) => {
+      return {
+        mlClient: mlClient.asScoped(request),
+      };
+    });
+
+    const routeInit: RouteInitialization = {
+      router: coreSetup.http.createRouter(),
+      getLicenseCheckResults: () => this.licenseCheckResults,
+    };
+
+    annotationRoutes(routeInit, plugins.security);
+    calendars(routeInit);
+    dataFeedRoutes(routeInit);
+    dataFrameAnalyticsRoutes(routeInit);
+    dataRecognizer(routeInit);
+    dataVisualizerRoutes(routeInit);
+    fieldsService(routeInit);
+    fileDataVisualizerRoutes(routeInit);
+    filtersRoutes(routeInit);
+    indicesRoutes(routeInit);
+    jobAuditMessagesRoutes(routeInit);
+    jobRoutes(routeInit);
+    jobServiceRoutes(routeInit);
+    notificationRoutes(routeInit);
+    resultsServiceRoutes(routeInit);
+    jobValidationRoutes(routeInit, this.version);
+    systemRoutes(routeInit, {
+      spacesPlugin: plugins.spaces,
+      cloud: plugins.cloud,
+    });
+    initMlServerLog({ log: this.log });
+    coreSetup.getStartServices().then(([core]) => {
+      makeMlUsageCollector(plugins.usageCollection, core.savedObjects);
+    });
+
+    plugins.licensing.license$.subscribe(async license => {
+      const { isEnabled: securityIsEnabled } = license.getFeature('security');
+      // @ts-ignore isAvailable is not read
+      const { isAvailable, isEnabled } = license.getFeature(this.pluginId);
+
+      this.licenseCheckResults = {
+        isActive: license.isActive,
+        // This `isAvailable` check for the ml plugin returns false for a basic license
+        // ML should be available on basic with reduced functionality (only file data visualizer)
+        // TODO: This will need to be updated in the second step of this cutover to NP.
+        isAvailable: isEnabled,
```

Comment:
How did this work before / what is the approach to solving this in the follow up?